bin/dynamic_challenge_score.py

```diff
@@ -0,0 +1,448 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import os
+import typing
+
+import tarfile
+import numpy as np
+from typing import Tuple, List
+import synapseclient
+
+
+def get_args():
```

Comment:
This might be a good opportunity to explore Click or Typer:

1. https://typer.tiangolo.com/
2. https://click.palletsprojects.com/en/8.1.x/

Reply:
Eventually I'd like to transition the Synapse Python Client to use one of these

Reply:
I love Typer, I implemented it for `agora-data-tools` a while ago, but I think for this use case it would be overkill. If this logic becomes part of a larger package at some point that is where I envision something like this coming in. Typically in Python scripts for Nextflow tooling we just pass positional arguments.